aws-s3/s3.mli

```diff
@@ -36,7 +36,9 @@ module Make(Io : Types.Io) : sig
   }
 
   type nonrec 'a result = ('a, error) result Deferred.t
-  type 'a command = ?credentials:Credentials.t -> ?connect_timeout_ms:int -> endpoint:Region.endpoint -> 'a
+
+  (* If [requester_pays], caller will pay AWS data transfer costs. *)
```

Comment:
This is not in line with the documentation. When operating on `Requester Pays buckets`, a special header must be present to indicate the callers willingness to accept charges. Setting the flag on non-requester pays bucket has no effect, and not setting the flag on requester-pays buckets, wil case any operation to fail (unless the caller owns the bucket).

Reply:
Maybe add a link to the [documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ObjectsinRequesterPaysBuckets.html) in the documentation comment for the flag.